internal/compress/gzip/gzip.go

```diff
@@ -0,0 +1,72 @@
+package gzip
+
+import (
+	"io"
+
+	"github.com/klauspost/compress/gzip"
+)
+
+// These constants are copied from the gzip package
+const (
+	NoCompression       = gzip.NoCompression
+	BestSpeed           = gzip.BestSpeed
+	BestCompression     = gzip.BestCompression
+	DefaultCompression  = gzip.DefaultCompression
+	ConstantCompression = gzip.ConstantCompression
+	HuffmanOnly         = gzip.HuffmanOnly
+)
+
+// ReaderWriter an interface to create Writer and Reader implementation.
+type ReaderWriter interface {
```

Comment:
I think this is better, what do you think?
```suggestion
type Builder interface {
```